src/loan.rs

```diff
@@ -590,18 +590,20 @@ impl Borrower0 {
                 )
             })?;
 
-        transaction
-            .output
-            .iter()
-            .find_map(|out| {
-                let unblinded_out = out.unblind(secp, self.address_blinding_sk).ok()?;
-                let is_collateral_change_out = unblinded_out.asset == self.bitcoin_asset_id
-                    && unblinded_out.value == collateral_change_amount.as_sat()
-                    && out.script_pubkey == self.address.script_pubkey();
+        if collateral_change_amount != Amount::ZERO {
+            transaction
+                .output
+                .iter()
+                .find_map(|out| {
+                    let unblinded_out = out.unblind(secp, self.address_blinding_sk).ok()?;
+                    let is_collateral_change_out = unblinded_out.asset == self.bitcoin_asset_id
+                        && unblinded_out.value == collateral_change_amount.as_sat()
+                        && out.script_pubkey == self.address.script_pubkey();
 
-                is_collateral_change_out.then(|| out)
-            })
-            .context("no collateral change txout")?;
+                    is_collateral_change_out.then(|| out)
+                })
+                .context("no collateral change txout")?;
```

Comment:
I would find this easier to understand if it were an explicit `if change_output.is_none()` with an `anyhow::bail!`.

Reply:
I'll give that a try!

Reply:
Tried to do this here: https://github.com/comit-network/baru/pull/58/commits/8d517c9eebaeb868eda30b39e11a5ad41317af66